sycl/include/sycl/vector_preview.hpp

```diff
@@ -983,13 +983,21 @@ template <typename Type, int NumElements> class vec {
     }
   }
 
-#ifdef __SYCL_BINOP
-#error "Undefine __SYCL_BINOP macro"
+#if defined(__SYCL_BINOP) || defined(BINOP_BASE)
+#error "Undefine __SYCL_BINOP and BINOP_BASE macro"
 #endif
 
-#ifdef __SYCL_USE_EXT_VECTOR_TYPE__
-#define __SYCL_BINOP(BINOP, OPASSIGN, CONVERT)                                 \
-  friend vec operator BINOP(const vec &Lhs, const vec &Rhs) {                  \
+// Note: vec<>/SwizzleOp logical value is 0/-1 logic, as opposed to 0/1 logic.
+// As far as CTS validation is concerned, 0/-1 logic also applies when
+// NumElements is equal to one, which is somewhat inconsistent with being
+// transparent with scalar data.
```

Comment:
I don't understand this comment. Could you explain it?

Reply:
This comment is from the current implementation (https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/vector.hpp#L1073), I just happen to re-arrange it.

IIUC, the comment states that the logical operations on scalar types results in 0/1, while for vec<>, logical operations should result in 0 and -1 (similar to OpenCL vectors). Here's what SPEC says:

> Each element of the SYCL vec that is returned must be -1 if the operation results in true and 0 if the operation results in false.

I can rewrite the comment to make it clearer.

Reply:
That would make sense! I do think a small rewrite could make it a little clearer.